src/Core/Services/Locator.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Locator.hpp"
+
+// Static variable default values
+std::unique_ptr<ArchHooks> Locator::archHooks = nullptr;
+std::unique_ptr<Core::ILogger> Locator::logger = nullptr;
+
+ArchHooks *Locator::getArchHooks() {
+    return archHooks.get();
+}
+
+Core::ILogger *Locator::getLogger() {
+    return logger.get();
+}
+
+void Locator::provide(ArchHooks *hooks) {
+    if(!Locator::archHooks)
+        Locator::archHooks = std::unique_ptr<ArchHooks>(hooks);
+}
+
+void Locator::provide(Core::ILogger *logger) {
+    if(!Locator::logger)
+        Locator::logger = std::unique_ptr<Core::ILogger>(logger);
```

Comment:
```suggestion
        Locator::logger = std::make_unique<Core::ILogger>(logger);
```

Reply:
View conversation in above review of Locator::archHooks.